src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-multisite-sync-policy-form/rgw-multisite-sync-policy-form.component.ts

```diff
@@ -0,0 +1,175 @@
+import { AbstractControl, AsyncValidatorFn, ValidationErrors, Validators } from '@angular/forms';
+import { ActivatedRoute, Router } from '@angular/router';
+import { Observable, timer as observableTimer, of } from 'rxjs';
+import {
+  catchError,
+  debounceTime,
+  distinctUntilChanged,
+  map,
+  mergeMap,
+  switchMapTo
+} from 'rxjs/operators';
+import { RgwBucketService } from '~/app/shared/api/rgw-bucket.service';
+import { RgwMultisiteService } from '~/app/shared/api/rgw-multisite.service';
+import { ActionLabelsI18n, URLVerbs } from '~/app/shared/constants/app.constants';
+import { NotificationType } from '~/app/shared/enum/notification-type.enum';
+import { CdFormBuilder } from '~/app/shared/forms/cd-form-builder';
+import { CdFormGroup } from '~/app/shared/forms/cd-form-group';
+import { NotificationService } from '~/app/shared/services/notification.service';
+import { RgwMultisiteSyncPolicyStatus } from '../models/rgw-multisite';
+import { CdForm } from '~/app/shared/forms/cd-form';
+import _ from 'lodash';
+
+@Component({
+  selector: 'cd-rgw-multisite-sync-policy-form',
+  templateUrl: './rgw-multisite-sync-policy-form.component.html',
+  styleUrls: ['./rgw-multisite-sync-policy-form.component.scss']
+})
+export class RgwMultisiteSyncPolicyFormComponent extends CdForm implements OnInit {
+  syncPolicyForm: CdFormGroup;
+  editing = false;
+  action: string;
+  resource: string;
+  syncPolicyStatus = RgwMultisiteSyncPolicyStatus;
+
+  bucketDataSource = (text$: Observable<string>) => {
+    return text$.pipe(
+      debounceTime(200),
+      distinctUntilChanged(),
+      mergeMap((token: string) => this.getBucketTypeahead(token))
+    );
+  };
+
+  constructor(
+    private router: Router,
+    private route: ActivatedRoute,
+    public actionLabels: ActionLabelsI18n,
+    private fb: CdFormBuilder,
+    private rgwMultisiteService: RgwMultisiteService,
+    private notificationService: NotificationService,
+    private rgwBucketService: RgwBucketService
+  ) {
+    super();
+    this.editing = this.router.url.startsWith(`/rgw/multisite/sync-policy/${URLVerbs.EDIT}`);
+    this.action = this.editing ? this.actionLabels.EDIT : this.actionLabels.CREATE;
+    this.resource = $localize`Sync Policy Group`;
+    this.createForm();
+    this.loadingReady();
+  }
+
+  ngOnInit(): void {
+    if (this.editing) {
+      this.route.paramMap.subscribe((params) => {
+        const groupName = params.get('groupName');
+        if (groupName) {
+          this.loadingStart();
+          this.rgwMultisiteService
+            .getSyncPolicy('', '', true)
+            .subscribe((syncPolicies: Array<Object>) => {
+              this.loadingReady();
+              if (syncPolicies && syncPolicies.length > 0) {
+                const syncPolicyToEdit: any = syncPolicies.find(
+                  (policy) => policy['id'] === decodeURIComponent(groupName)
+                );
+                if (syncPolicyToEdit) {
```

Comment:
rather than getting all the policies again and iterating through it, we can get a single policy since the group name is already known right?
https://github.com/ceph/ceph/blob/main/src/pybind/mgr/dashboard/controllers/rgw.py#L152

maybe you'll need the bucket name in some cases but that's something that you can send it through router parms from the other page (same way you are gettng the groupName`

Reply:
sure, I will work upon this one